stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/extensions/filter.py

```diff
@@ -210,3 +232,22 @@ def to_es(arg: Arg):
         return arg
     else:
         raise RuntimeError(f"unknown arg {repr(arg)}")
+
+
+def cql2_like_to_es(input_string):
+    """Convert arugument in CQL2 ('_' and '%') to Elasticsearch wildcard operators ('?' and '*', respectively). Handle escape characters and handle Elasticsearch wildcards directly."""
+    es_string = ""
+    escape = False
```

Comment:
this is always False everywhere it's read, and it's modified but never read or returned after  that

Reply:
`escape` is set to `True` when the for loop encounters an escape character (`/`). It is set to false at the end of the next loop iteration so the character after a `/` is not replaced with a `?` or `*`. 

The code is functional. But I am open to suggestions for a better way to handle escape characters.

Reply:
Replaced logic with regex.